### ISSUE TITLE
Suppress benign JIT build warnings

### ISSUE THREAD

**Issue by magnumripper** [hashcat/hashcat#145]
Some OpenCL platforms are more chatty than others when building kernels. Sometimes it's a good thing as you can get a hint about problems with the code, and sometimes it's just a nuisance.

A simple way to handle this is to mute the build output unless build actually failed. But for developers this will hide bugs so you'd probably want a way to override it (eg. a -v / --verbose option) so you can see mere warnings too. Not sure what is best. In JtR you need to bump verbosity to see them _if running a release build_ while you will always see them if not (or if the build actually fails).

Example: Here's OSX running siphash with no cached kernel:

```
$ ./oclHashcat.app -b -d3 -m10100
oclHashcat v2.10 starting in benchmark-mode...

Device #1: Intel(R) Core(TM) i7-3615QM CPU @ 2.30GHz, skipped
Device #2: HD Graphics 4000, skipped
Device #3: GeForce GT 650M, 256/1024 MB allocatable, 900Mhz, 2MCU

<program source>:95:26: warning: comparison of integers of different signs: 'int' and 'const u32' (aka 'const unsigned int')
    for (i = 8, j = 2; i <= pw_len; i += 8, j += 2)
                       ~ ^  ~~~~~~
<program source>:196:26: warning: comparison of integers of different signs: 'int' and 'const u32' (aka 'const unsigned int')
    for (i = 8, j = 2; i <= pw_len; i += 8, j += 2)
                       ~ ^  ~~~~~~
In file included from <program source>:18:
./OpenCL/common.c:71:13: warning: unused function 'truncate_block'
static void truncate_block (u32 w[4], const u32 len)
            ^
./OpenCL/common.c:134:13: warning: unused function 'make_unicode_S'
static void make_unicode_S (const u32 in[4], u32 out1[4], u32 out2[4])
            ^
./OpenCL/common.c:159:13: warning: unused function 'make_unicode'
static void make_unicode (const u32x in[4], u32x out1[4], u32x out2[4])
            ^
./OpenCL/common.c:184:13: warning: unused function 'undo_unicode_S'
static void undo_unicode_S (const u32 in1[4], const u32 in2[4], u32 out[4])
            ^
./OpenCL/common.c:205:13: warning: unused function 'undo_unicode'
static void undo_unicode (const u32x in1[4], const u32x in2[4], u32x out[4])
            ^
./OpenCL/common.c:226:13: warning: unused function 'append_0x01_1x4'
static void append_0x01_1x4 (u32 w0[4], const u32 offset)
            ^
./OpenCL/common.c:296:13: warning: unused function 'append_0x01_2x4'
static void append_0x01_2x4 (u32 w0[4], u32 w1[4], const u32 offset)
            ^
./OpenCL/common.c:430:13: warning: unused function 'append_0x01_3x4'
static void append_0x01_3x4 (u32 w0[4], u32 w1[4], u32 w2[4], const u32 offset)
            ^
./OpenCL/common.c:628:13: warning: unused function 'append_0x01_4x4'
static void append_0x01_4x4 (u32 w0[4], u32 w1[4], u32 w2[4], u32 w3[4], const u32 offset)
            ^
./OpenCL/common.c:890:13: warning: unused function 'append_0x01_8x4'
static void append_0x01_8x4 (u32 w0[4], u32 w1[4], u32 w2[4], u32 w3[4], u32 w4[4], u32 w5[4], u32 w6[4], u32 w7[4], const u32 offset)
            ^
./OpenCL/common.c:1408:13: warning: unused function 'append_0x02_1x4'
static void append_0x02_1x4 (u32 w0[4], const u32 offset)
            ^
./OpenCL/common.c:1478:13: warning: unused function 'append_0x02_2x4'
static void append_0x02_2x4 (u32 w0[4], u32 w1[4], const u32 offset)
            ^
./OpenCL/common.c:1612:13: warning: unused function 'append_0x02_3x4'
static void append_0x02_3x4 (u32 w0[4], u32 w1[4], u32 w2[4], const u32 offset)
            ^
./OpenCL/common.c:1810:13: warning: unused function 'append_0x02_4x4'
static void append_0x02_4x4 (u32 w0[4], u32 w1[4], u32 w2[4], u32 w3[4], const u32 offset)
            ^
./OpenCL/common.c:2072:13: warning: unused function 'append_0x02_8x4'
static void append_0x02_8x4 (u32 w0[4], u32 w1[4], u32 w2[4], u32 w3[4], u32 w4[4], u32 w5[4], u32 w6[4], u32 w7[4], const u32 offset)
            ^
./OpenCL/common.c:2590:13: warning: unused function 'append_0x80_1x4'
static void append_0x80_1x4 (u32 w0[4], const u32 offset)
            ^
./OpenCL/common.c:2660:13: warning: unused function 'append_0x80_2x4'
static void append_0x80_2x4 (u32 w0[4], u32 w1[4], const u32 offset)
            ^
./OpenCL/common.c:2794:13: warning: unused function 'append_0x80_3x4'
static void append_0x80_3x4 (u32 w0[4], u32 w1[4], u32 w2[4], const u32 offset)
            ^
./OpenCL/common.c:2992:13: warning: unused function 'append_0x80_4x4'
static void append_0x80_4x4 (u32 w0[4], u32 w1[4], u32 w2[4], u32 w3[4], const u32 offset)
            ^
./OpenCL/common.c:3254:13: warning: unused function 'append_0x80_8x4'
static void append_0x80_8x4 (u32 w0[4], u32 w1[4], u32 w2[4], u32 w3[4], u32 w4[4], u32 w5[4], u32 w6[4], u32 w7[4], const u32 offset)
            ^
./OpenCL/common.c:3772:13: warning: unused function 'append_0x80_1x16'
static void append_0x80_1x16 (u32 w[16], const u32 offset)
            ^
./OpenCL/common.c:4034:13: warning: unused function 'switch_buffer_by_offset_S'
static void switch_buffer_by_offset_S    (u32 w0[4], u32 w1[4], u32 w2[4], u32 w3[4], const u32 offset)
            ^
./OpenCL/common.c:4748:13: warning: unused function 'switch_buffer_by_offset_be_S'
static void switch_buffer_by_offset_be_S (u32 w0[4], u32 w1[4], u32 w2[4], u32 w3[4], const u32 offset)
            ^
./OpenCL/common.c:5253:13: warning: unused function 'switch_buffer_by_offset'
static void switch_buffer_by_offset    (u32x w0[4], u32x w1[4], u32x w2[4], u32x w3[4], const u32 offset)
            ^
./OpenCL/common.c:5967:13: warning: unused function 'switch_buffer_by_offset_be'
static void switch_buffer_by_offset_be (u32x w0[4], u32x w1[4], u32x w2[4], u32x w3[4], const u32 offset)
            ^

Hashtype: SipHash
Workload: 1024 loops, 256 accel

Speed.Dev.#3.:   731.7 MH/s
```

**Comment by jsteube** [comment]
I made the same experience with the different OpenCL platforms. But I think it's the opposite direction we should go here:

1) As sad as it sounds, I made the experience that it's the normal users that report back such warnings instead of the beta testers. I think it's because they're scared by it much more. Therefore I think it's better to no surpress them for the release version

2) Such a verbose switch isn't too bad. Reason here is that if t's always the same warnings they becomes too "normal" and people (me included) tend to ignore/overlook them. This is bad because then it's hard to notice new warnings that could come up while changing the code which are actually important
